music-rpc.ts

```diff
@@ -46,6 +49,14 @@ class AppleMusicDiscordRPC {
 
   async setActivityLoop(): Promise<void> {
     try {
+      // Check if Discord is running
+      const discordRunning = await isAnyDiscordRunning();
+      if (!discordRunning) {
+        console.log("No Discord client is running. Waiting before retry...");
+        return;
+      }
+    
+    // Try to connect
```

Comment:
```suggestion
    // Check if Discord is running
    const discordRunning = await isAnyDiscordRunning();
    if (!discordRunning) {
      console.log("No Discord client is running. Waiting before retry...");
      return;
    }

    // Try to connect
    try {
```
Move the check out of the try block